src/main/java/com/iotiq/application/service/GameService.java

```
@@ -52,4 +56,48 @@ private Game mapRequestToEntity(CreateGameRequest request) {
         // game.setAudioFileUrl(audioFileUrl());
         return game;
     }
+
+    public void updateGameStatus(UUID id, UpdateGameStatusRequest request) {
+        Game game = gameRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("game"));
+
+        // TODO: Calculate game time if status transitions from PAUSED to IN_PROGRESS.
+        game.setStatus(request.status());
+        gameRepository.save(game);
+    }
+
+    @Transactional
+    public QuestionResponse getNextQuestion(UUID gameId) {
+        // Retrieve the currently logged-in user
+        User user = userService.getCurrentUser();
+        // Get or create a Gamer entity associated with the user
+        Gamer gamer = gamerService.getOrCreateGamerForUser(user);
+
+        // Validation checks
+        Game game = gameRepository.findById(gameId).orElseThrow(() -> new EntityNotFoundException("game"));
+        validateGameInProgress(game);
+
+        // Get the next question
+        QuestionDto questionDto = questionService.getNextQuestion();
+        
+        // If there are no more questions, end the game
+        if (questionDto == null) {
+            game.setStatus(GameStatus.ENDED);
+            return null;
+        }
```

We can throw an error when the game ends here instead of returning null.